bluemira/geometry/parameterisations.py

```diff
@@ -1016,34 +1014,63 @@ def _make_domed_leg(
 
         Parameters
         ----------
-        axis: np.ndarray
-            [x,y,z] vector normal to plane of parameterisation
-        x_out: np.float
+        x_out: float
             Radial position of outer edge of limb [m]
-        x_curve start: np.float
+        x_curve start: float
             Radial position of straight-curve transition of limb [m]
-        x_mid: np.float
+        x_mid: float
             Radial position of inner edge of  upper/lower limb [m]
-        z_top: np.float
+        z_top: float
             Vertical position of top of limb dome [m]
-        z_mid: np.float
+        z_mid: float
             Vertical position of flat section [m]
+        r_c: float
+            Radius of corner transition. Nominally 0 [m]
+        axis: np.ndarray
+            [x,y,z] vector normal to plane of parameterisation
         flip: bool
             True if limb is lower limb of section, False if upper
-        r_c: np.float
-            Radius of corner transition. Nominally 0 [m]
 
         Returns
         -------
         shape: BluemiraWire
             CAD Wire of the geometry
         """
+        # Labels
+        if flip:
+            label = "bottom"
+            z_top_r_leg = lambda: z_top + r_leg
```

Comment:
Most of them are collected because the variables in question are defined later in the code otherwise `if flip` is peppered throughout the function. I didnt like it before and I'm not really sure on it now but I didnt come up with a nicer way to collect all the variations together

Reply:
Ah, I see now - some variables used in the `lambda`s do not exist until later on. This is fine with me, just unusual to see `lambda`s like this. Potentially worth a comment explaining why they're used ("_some variables used within these lambdas are undefined at this point, using functions saves having to use `if flip .. else ..` throughout the function_"), may save some head-scratching down-the-line.

Probably not worth the time, but some potential solutions are: to have two separate functions `_make_domed_leg_bottom` and `_make_domed_leg_top` with common code refactored into helper functions, then the only `if flip` would be deciding which of the two methods to call; or just always build a "bottom" then perform a rotation after the wire is built `if flip` (I'm not familiar enough with the code to be sure how feasible that is). I'll leave it up to you to decide what you think's best/worth doing, feel free to resolve this comment when you're happy.

Reply:
Done here (not it the meta class though)